src/components/manage/Blocks/Listing/ListingBody.jsx

```diff
@@ -14,29 +14,37 @@ import paginationRightSVG from '@plone/volto/icons/right-key.svg';
 const ListingBody = React.memo(
   (props) => {
     const { data, properties, path, isEditMode, variation } = props;
-    const [currentPage, setCurrentPage] = React.useState(1);
     const content = properties;
+    const { settings } = config;
+    const { batch_size = settings.defaultPageSize } = data;
+
+    const adaptedQuery = {
+      ...(data.limit ? { limit: data.limit } : {}),
+      ...(data.query ? { query: data.query } : {}),
+      b_size: batch_size,
+      fullobjects: 1,
+    };
```

Comment:
I'm not a fan of rewriting parameters for the query .. because some parameters may be forgotten ..
for example, **sort_on** and **sort_order** are missing here .. and maybe some other params that i'm not thinking in this moment also i fixed the b_size param here in this pr: https://github.com/plone/volto/pull/2479, maybe you colud improve that pr adding your default settings.defaultPageSize